pubsub/google/cloud/pubsub_v1/subscriber/message.py

```diff
@@ -162,6 +163,30 @@ def ack_id(self):
         """str: the ID used to ack the message."""
         return self._ack_id
 
+    @property
+    def delivery_attempt(self):
+        """The delivery attempt counter is 1 + (the sum of number of NACKs
+        and number of ack_deadline exceeds) for this message. It is set to zero
```

Comment:
This was brought up in an email today. Should we set it to None when dead letter policy is not set?

Reply:
Having zero delivery attempts indeed feels odd, and `None` would represent the N/A (not applicable) semantics better, even if the backend denotes the same with a zero.

A side effect of returning `None` instead would also be forcing the other code to explicitly handle that case, rather getting away with simply saying `if msg.delivery_attempt > some_int`, for example. The latter could result in a silent bug, while throwing `None` into the mix would blow up loudly under Python 3.

Reply:
It was decided just a day or two ago to go with the None option. Will make the change.

Reply:
Let me know when I can review the nil change.

Reply:
Made the changes; please take a look.